File: examples/java/suspend/DreamMaster.java
/* Copyright (c) 2006-2014. The SimGrid Team.
 * All rights reserved.                                                     */

/* This program is free software; you can redistribute it and/or modify it
 * under the terms of the license (GNU LGPL) which comes with this package. */

package suspend;
import org.simgrid.msg	.Host;
import org.simgrid.msg.Msg;
import org.simgrid.msg.Process;
import org.simgrid.msg.MsgException;
import org.simgrid.msg.HostNotFoundException;

public class DreamMaster extends Process {
  public DreamMaster(String hostname, String name)  throws HostNotFoundException{
    super(hostname,name);
  }

  public void main(String[] args) throws MsgException {
    Msg.info("Let's create a lazy guy.");
    Process lazyGuy = new LazyGuy(getHost(),"Lazy",null);
    lazyGuy.start();
    Msg.info("Let's wait a little bit...");
    waitFor(10);
    Msg.info("Let's wake the lazy guy up! >:) BOOOOOUUUHHH!!!!");
    lazyGuy.resume();
    Msg.info("OK, goodbye now.");
  }
}

File: examples/cxx/suspend/DreamMaster.cxx
#include "DreamMaster.hpp"
#include "LazyGuy.hpp"
#include <Host.hpp>
#include <HostNotFoundException.hpp>
#include <Msg.hpp>

#include <Msg.hpp>


MSG_IMPLEMENT_DYNAMIC(DreamMaster, Process);

int DreamMaster::main(int argc, char** argv)
{
	
	info("Hello");
	
	info("Let's create a lazy guy.");
	

	LazyGuy* lazy;

	try
	{
		Host currentHost = Host::currentHost();
	
		info(TEXT_("Current host  name : ") + TEXT_(currentHost.getName()));
	
		lazy = new LazyGuy(currentHost,"LazyGuy");
	
		info("Let's wait a little bit...");
	
		Process::sleep(10.0);
	
		info("Let's wake the lazy guy up! >:) ");
	
		lazy->resume();
	
	
	}
	catch(HostNotFoundException e)
	{
		error(TEXT_(e.toString()) + TEXT_(". Stopping Now!"));
		exit(1);
	}
	
	//lazy->migrate(currentHost);
	
	info("OK, goodbye now.");

	delete this;

	return 0;
	
}